### DISCUSSION TITLE
Dots in table names?

### DISCUSSION THREAD

**Discussion by rsund** [duckdb/duckdb#6312] [Q&A] [answered]
As can be seen from the following example,
```
CREATE OR REPLACE SCHEMA test;
CREATE OR REPLACE TABLE test.tbl AS SELECT 1;
CREATE OR REPLACE TABLE "test.tbl" AS SELECT 2;
FROM information_schema.tables;
```
two different tables will be created. The first one is in the test-schema as expected, but the second one is actually in the main-schema having a table name containing dot.

This may be intended behaviour, but woudn't it be less confusing, if those both worked likewise? Probably that could be achieved quite easily if the second case would be automatically converted from `"test.tbl"` to `"test"."tbl"` while parsing the table name. Would that make any sense?

**Comment by rsund** [comment]
I encountered this problem in the R client ("table" name is quoted by a helper function). Please find below a reprex. This could be fixed in the R client, but the suggestion above would be a more general solution. 

``` r
library(duckdb)
#> Loading required package: DBI
df <- data.frame(a=1:10)
con <- DBI::dbConnect(duckdb::duckdb())
DBI::dbExecute(con, "CREATE SCHEMA test")
#> [1] 0
DBI::dbWriteTable(con, "test.tbl", df) # This is working incorrectly

duckdb::duckdb_register(con, "dif", df)
DBI::dbExecute(con, "CREATE TABLE test.tbl AS FROM dif")
#> [1] 10
duckdb::duckdb_unregister(con, "dif")

DBI::dbGetQuery(con, "FROM information_schema.tables")
#>   table_catalog table_schema table_name table_type self_referencing_column_name
#> 1        memory         main   test.tbl BASE TABLE                           NA
#> 2        memory         test        tbl BASE TABLE                           NA
#>   reference_generation user_defined_type_catalog user_defined_type_schema
#> 1                   NA                        NA                       NA
#> 2                   NA                        NA                       NA
#>   user_defined_type_name is_insertable_into is_typed commit_action
#> 1                     NA                YES       NO          <NA>
#> 2                     NA                YES       NO          <NA>
DBI::dbDisconnect(con, shutdown=TRUE)
```

<sup>Created on 2023-02-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by Mytherin** [comment, marked as answer]
It is intentional that dots are ignored when quoted (and is part of the SQL standard). We could fix this separately in the R client, either by parsing the dots or by providing additional parameters to functions like `dbWriteTable`.

**Reply by rsund** [reply]
Thanks for the clarification - I totally agree that (a potential) fix in the R client would be enough.
